src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/DefineOutputType.cs

```diff
@@ -0,0 +1,23 @@
+﻿namespace DotNetProjectFile.Analyzers.MsBuild;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+public sealed class DefineOutputType : MsBuildProjectFileAnalyzer
+{
+    public DefineOutputType() : base(Rule.DefineOutputType) { }
+
+    protected override void Register(ProjectFileAnalysisContext context)
+    {
+        if (context.Project.IsProject)
+        {
+            var outputTypes = context.Project
+                .AncestorsAndSelf()
```

Comment:
I would argue that it than also should not be a an inherited setting.

Reply:
I would argue that it should be. I personally like to simply create different props files for different purposes (e.g. 1 for packages and 1 for tests) that each specify the `OutputType` for their type. Of course enforcing that there is only 1 definition would be very useful, but that is related to #15 .

Reply:
Also a fair point. But in that case for a property like this (and also for `<TargetFramework>` or `<IsTestProject>`) you could/should argue that those properties should then not be overridden: if defined in a prop, leave it as it. I'm not sure yet if that should be part of this rule (for `<RargetOutput>`) or not.